app/lib/features/chat/widgets/rooms_list.dart

```diff
@@ -11,25 +13,174 @@ import 'package:flutter_svg/flutter_svg.dart';
+    );
+  }
+
+  Widget filterBox(BuildContext context) {
+    final selected =
+        ref.watch(roomListFilterProvider.select((value) => value.selection));
+    final hasSearchTerm = ref
+            .watch(roomListFilterProvider.select((value) => value.searchTerm))
+            ?.isNotEmpty ==
+        true;
+    final hasFilters = ref.watch(hasRoomFilters);
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.start,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        const SizedBox(height: 5),
+        SearchBar(
+          focusNode: searchFocus,
+          controller: searchTextController,
+          leading: const Icon(Atlas.magnifying_glass),
+          hintText: 'Search chats',
+          trailing: hasSearchTerm
+              ? [
+                  InkWell(
+                    onTap: () {
+                      searchTextController.clear();
+                      ref
+                          .read(roomListFilterProvider.notifier)
+                          .updateSearchTerm(null);
+                    },
+                    child: const Icon(Icons.clear),
+                  ),
+                ]
+              : null,
+          onChanged: (value) {
+            ref.read(roomListFilterProvider.notifier).updateSearchTerm(value);
+          },
+        ),
+        const SizedBox(height: 20),
+        SegmentedButton<FilterSelection>(
+          segments: const [
+            ButtonSegment(
+              value: FilterSelection.all,
+              label: Text('all'),
+            ),
+            ButtonSegment(
+              value: FilterSelection.favorites,
+              icon: Icon(Icons.bookmark_outline),
+            ),
+            ButtonSegment(
+              value: FilterSelection.dmsOnly,
+              label: Text("only DM's"),
+            ),
+          ],
+          onSelectionChanged: (Set<FilterSelection> newSelection) async {
+            await ref
+                .read(roomListFilterProvider.notifier)
+                .setSelection(newSelection.first);
+          },
+          selected: {selected},
+        ),
+        const SizedBox(height: 10),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: [
+            if (hasFilters)
+              OutlinedButton(
+                onPressed: () {
+                  searchTextController.clear();
+                  ref.read(roomListFilterProvider.notifier).clear();
+                  setState(() {
+                    _isSearchVisible = false;
+                  });
+                },
+                child: const Text('clear'),
+              ),
+            if (!hasFilters)
+              OutlinedButton(
+                onPressed: () {
+                  setState(() {
+                    _isSearchVisible = false;
+                  });
+                },
+                child: const Text('close'),
+              ),
+          ],
+        ),
```

Comment:
This `Clear / Close` UX Flow can be improved. Also on UI side, there are a lot spacing on right side of the button.

![Screenshot 2024-03-04 at 1 57 09 PM](https://github.com/acterglobal/a3/assets/51526480/29e273e1-42a5-4d4e-b326-e66a6d22c850)

Reply:
> This `Clear / Close` UX Flow can be improved.

How would you like it to be improved?